src/ApiConnectors/BaseApiConnector.php

```diff
@@ -42,9 +56,53 @@ abstract class BaseApiConnector implements LoggerAwareInterface
      */
     private $numRetries = 0;
 
-    public function __construct(AuthenticatedConnection $connection)
+    /**
+     * @var mixed
+     */
+    private $config;
+
+    public function __construct(AuthenticatedConnection $connection, array $configuration = [])
     {
         $this->connection = $connection;
+        $this->setConfig($configuration);
+    }
+
+    /**
+     * @param array $configuration
+     */
+    private function setConfig(array $configuration): void
+    {
+
+        if (!empty($configuration[self::CONFIG_MAX_RETRIES]) && (int) $configuration[self::CONFIG_MAX_RETRIES] >= 0) {
```

Comment:
I'd prefer explicit checking of the validity of the value and throwing an `InvalidArgumentException` instead of always ignoring if int casted its not `>= 0`

Reply:
done